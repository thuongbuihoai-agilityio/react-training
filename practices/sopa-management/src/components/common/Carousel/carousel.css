.carousel {
  position: relative;
  width: var(--w-extra-large);
  margin: var(--m-3xl) auto var(--m-extra-large);
}

.carousel-inner {
  position: relative;
  overflow: hidden;
  width: 100%;
}

.carousel-info {
  position: absolute;
  bottom: 40px;
  left: 40px;
}

.carousel-heading {
  color: var(--light);
  font-family: 'var(--font-secondary)';
  font-weight: var(--font-bold);
  line-height: 74px;
  margin-bottom: var(--m-xl);
  width: 570px;
}

.carousel-open:checked + .carousel-item {
  position: static;
  opacity: 1;
}

.carousel-item {
  position: absolute;
  opacity: 0;
  -webkit-transition: opacity 0.6s ease-out;
  transition: opacity 0.6s ease-out;
}

.carousel-item img {
  display: block;
  border-radius: 16px;
  width: 1170px;
  height: 657px;
}

.carousel-control {
  border-radius: 50%;
  color: var(--light);
  cursor: pointer;
  display: none;
  font-size: var(--text-medium);
  height: 40px;
  line-height: 35px;
  position: absolute;
  top: 50%;
  cursor: pointer;
  text-align: center;
  width: 40px;
  z-index: 10;
}

.carousel-indicators {
  list-style: none;
  margin: 0;
  padding: 0;
  position: absolute;
  bottom: -10px;
  left: 0;
  right: 0;
  text-align: center;
  z-index: 10;
}

.carousel-indicators li {
  display: inline-block;
  margin: 0 var(--m-small);
}

.carousel-bullet {
  display: inline-block;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background-color: var(--light);
  margin: 0 1px var(--m-extra-normal);
  cursor: pointer;
}

#carousel-1:checked ~ .carousel-indicators li:nth-child(1) .carousel-bullet,
#carousel-2:checked ~ .carousel-indicators li:nth-child(2) .carousel-bullet,
#carousel-3:checked ~ .carousel-indicators li:nth-child(3) .carousel-bullet {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background-color: var(--dark);
}
