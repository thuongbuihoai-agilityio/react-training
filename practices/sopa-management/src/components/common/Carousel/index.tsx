// Images
import { IMAGE } from '@constants/image';

// Components
import {
  Button,
  ButtonType,
  Image,
  Input,
  SizeType,
  Text,
} from '@components/common';

// Constants
import { CAROUSEL_BULLET } from '@constants/common';

// Styles
import './carousel.css';

const Carousel = () => (
  <div data-testid='carousel' className='carousel'>
    <div className='carousel-inner'>
      <Input
        className='carousel-open'
        type='radio'
        id='carousel-1'
        name='carousel'
        hidden
        checked
      />
      <figure className='carousel-item'>
        <Image
          url={IMAGE.banner}
          alt={IMAGE.altBanner}
          width={1170}
          height={657}
        />
      </figure>
      <Input
        className='carousel-open'
        type='radio'
        id='carousel-2'
        name='carousel'
        hidden
      />
      <figure className='carousel-item'>
        <Image
          url={IMAGE.banner2}
          alt={IMAGE.altBanner}
          width={1170}
          height={657}
          size='50vw'
        />
      </figure>
      <Input
        className='carousel-open'
        type='radio'
        id='carousel-3'
        name='carousel'
        hidden
      />
      <figure className='carousel-item'>
        <Image
          url={IMAGE.banner}
          alt={IMAGE.altBanner}
          width={1170}
          height={657}
        />
      </figure>
      <ul className='carousel-indicators'>
        {CAROUSEL_BULLET.map((item) => (
          <li key={item.key}>
            <label htmlFor={item.htmlFor} className={item.className} />
          </li>
        ))}
      </ul>
    </div>
    <div className='carousel-info'>
      <Text
        text='Step inside, for comfort and magic await you.'
        type={SizeType.large}
        className='carousel-heading'
      />
      <Button
        ariaLabel='Show now'
        children='Show now'
        type={ButtonType.primary}
      />
    </div>
  </div>
);

export default Carousel;
