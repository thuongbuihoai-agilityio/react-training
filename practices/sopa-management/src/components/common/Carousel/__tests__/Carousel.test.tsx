// Libs
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

// Components
import Carousel from '@common/Carousel';

describe('Carousel component', () => {
  test('should render Carousel component', () => {
    const { getByTestId } = render(<Carousel />);
    expect(getByTestId('carousel')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(<Carousel />);
    expect(container).toMatchSnapshot();
  });
});
