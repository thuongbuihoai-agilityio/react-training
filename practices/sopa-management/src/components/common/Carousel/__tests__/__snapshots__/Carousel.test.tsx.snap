// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component matches snapshot 1`] = `
<div>
  <div
    class="carousel"
    data-testid="carousel"
  >
    <div
      class="carousel-inner"
    >
      <input
        aria-hidden="true"
        checked=""
        class="carousel-open input input-"
        data-testid="input"
        hidden=""
        id="carousel-1"
        name="carousel"
        type="radio"
        value=""
      />
      <figure
        class="carousel-item"
      >
        <img
          alt="This is banner"
          class=""
          data-testid="image"
          src="/images/banner.webp"
        />
      </figure>
      <input
        aria-hidden="true"
        class="carousel-open input input-"
        data-testid="input"
        hidden=""
        id="carousel-2"
        name="carousel"
        type="radio"
        value=""
      />
      <figure
        class="carousel-item"
      >
        <img
          alt="This is banner"
          class=""
          data-testid="image"
          src="/images/banner-2.webp"
        />
      </figure>
      <input
        aria-hidden="true"
        class="carousel-open input input-"
        data-testid="input"
        hidden=""
        id="carousel-3"
        name="carousel"
        type="radio"
        value=""
      />
      <figure
        class="carousel-item"
      >
        <img
          alt="This is banner"
          class=""
          data-testid="image"
          src="/images/banner.webp"
        />
      </figure>
      <ul
        class="carousel-indicators"
      >
        <li>
          <label
            class="carousel-bullet"
            for="carousel-1"
          />
        </li>
        <li>
          <label
            class="carousel-bullet"
            for="carousel-2"
          />
        </li>
        <li>
          <label
            class="carousel-bullet"
            for="carousel-3"
          />
        </li>
      </ul>
    </div>
    <div
      class="carousel-info"
    >
      <p
        class="carousel-heading text-large"
        data-testid="text"
      >
        Step inside, for comfort and magic await you.
      </p>
      <button
        aria-label="Show now"
        class=" btn btn-primary"
        data-testid="button"
      >
        Show now
      </button>
    </div>
  </div>
</div>
`;
